lib/reek/smells/smell_detector.rb

```diff
@@ -104,16 +93,23 @@ def examine(context)
         self.smells_found += sm
       end
 
-      def enabled_for?(context)
-        enabled? && config_for(context)[SmellConfiguration::ENABLED_KEY] != false
+      def report_on(report)
+        smells_found.each { |smell| smell.report_on(report) }
       end
 
       def exception?(context)
         context.matches?(value(EXCLUDE_KEY, context, DEFAULT_EXCLUDE_SET))
       end
 
-      def report_on(report)
-        smells_found.each { |smell| smell.report_on(report) }
+      protected
+
+      # NOTE: Writers cannot be private in Ruby < 2.2
+      attr_writer :smells_found
+
+      private
+
+      def enabled_for?(context)
+        config.enabled? && config_for(context)[SmellConfiguration::ENABLED_KEY] != false
```

Comment:
Arrgh, i had written this comment while commuting on my phone and apparently I slipped some lines down :)

See my new comment above: https://github.com/troessner/reek/pull/651/files#diff-5b6de37fbe6e3791a90ceaf86e847935R75